eppo_core/src/event_ingestion/event_ingestion.rs

```diff
@@ -60,17 +60,28 @@ impl EventIngestionConfig {
 /// A handle to Event Ingestion subsystem.
 pub struct EventIngestion {
     tx: mpsc::Sender<BatchedMessage<Event>>,
+    context_sender: mpsc::Sender<(String, ContextValue)>,
 }
 
 impl EventIngestion {
     /// Starts the event ingestion subsystem on the given background runtime.
     pub fn spawn(runtime: &BackgroundRuntime, config: &EventIngestionConfig) -> EventIngestion {
-        let event_delivery = EventDelivery::new(
+        let event_delivery = Arc::new(Mutex::new(EventDelivery::new(
             reqwest::Client::new(),
             config.sdk_key.clone(),
             config.ingestion_url.clone(),
-        );
-
+        )));
+
+        let event_delivery_clone = Arc::clone(&event_delivery);
+        let (context_sender, mut context_rx) = mpsc::channel::<(String, ContextValue)>(100);
```

Comment:
Using a separate channel for context does not guarantee ordering and may suffer from partial writes (which is probably unexpected by users)

Given the following piece of user code:
```ruby
client.set_context("field1", "hello")
client.set_context("field2", "world")
client.track(event)
```
The event might get sent out with only `field1` in the context (but not `field2`), or with no context at all.